site/src/App.tsx

```diff
@@ -15,28 +17,81 @@ import AdminPage from './pages/AdminPage';
 import ReviewsPage from './pages/ReviewsPage';
 import SideBar from './component/SideBar/SideBar';
 
+import ThemeContext from './style/theme-context';
+import axios from 'axios';
+import { useCookies } from 'react-cookie';
+
 export default function App() {
+  // default darkMode to local or system preferences
+  const [darkMode, setDarkMode] = useState(
+    localStorage.getItem('theme') === 'dark'
+      ? true
+      : localStorage.getItem('theme') === 'light'
+        ? false
+        : window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches,
```

Comment:
not sure what I think of having nested ternaries. I think we could do something like this, but not sure how readable this is either 😅 

```ts
const localDark = ({ dark: true, light: false})[localStorage.theme];
const mediaDark = window.matchMedia?.('(prefers-color-scheme: dark)')?.matches
// ...
useState(localDark ?? mediaDark)
```

Reply:
I'll refactor to use a function for both occurrences of this ternary.